### PR TITLE
[NVIDIA] Support new config option `USE_FP8`

### PR DESCRIPTION
**Pull request by kaixih**
This PR introduces a new performance-related config option `USE_FP8`, which will call the provided function `tr_set_fp8_quantization` in praxis to set the recommended layers inside the transformer to utilize the FP8 GEMM.

There are four related PRs, and should be reviewed in this order:
(1) https://github.com/google/praxis/pull/29
(2) https://github.com/google/paxml/pull/48
(3) https://github.com/google/praxis/pull/28
current-->(4) https://github.com/google/paxml/pull/49

cc. @pjannaty @reedwm @nluehr @lukaszlew

### CODE REVIEW

File: paxml/trainer_lib.py [hunk 1]
```diff
@@ -1130,8 +1130,15 @@ def train_step_single_learner(
     wps_with_opt = tasks_lib.filter_vars_for_grad_or_opt(
         var_weight_hparams, excluded_for_opt
     )
```
Comment (lukaszlew) [inline paxml/trainer_lib.py:1132]: should this be in this PR?
(maybe diff was set incorrectly for a sequential PRs)?

**Review by lukaszlew — CHANGES_REQUESTED** [review]

File: paxml/trainer_lib.py (hunk 1 above)
Comment (kaixih) [reply paxml/trainer_lib.py:1132]: No, this PR is based on [the first PR](https://github.com/google/paxml/pull/48). This PR is essentially only this file: paxml/tasks/lm/model_params.py. Anyway, I think I will remove this PR and merge it to the first PR.

File: paxml/tasks/lm/model_params.py [hunk 2]
```diff
@@ -619,6 +621,10 @@ def task(self) -> pax_fiddle.Config[tasks_lib.SingleTask]:
     transformer_layer_p.norm_policy = self.NORM_POLICY
     transformer_layer_p.tr_atten_tpl.use_bias = False
     transformer_layer_p.tr_atten_tpl.combine_qkv = self.COMBINE_QKV
+
+    if self.USE_FP8:
+      fp8_ops.tr_set_fp8_quantization(transformer_layer_p)
```
Comment (lukaszlew) [inline paxml/tasks/lm/model_params.py:626]: PTAL the comment on the other PR.

**Review by lukaszlew — APPROVED** [review]

**Review by zhangqiaorjc — APPROVED** [review]

**Comment by kaixih** [comment]
@zhangqiaorjc I've noticed that this pull request has been in a "pull ready" status for a couple of days. Is there a specific action needed, like clicking a button to merge the PR, or will the process be automated by a robot?

**Pull request merged**